pkg/controller/core.oam.dev/v1alpha2/application/revision.go

```diff
@@ -856,6 +856,9 @@ func (h *AppHandler) FinalizeAndApplyAppRevision(ctx context.Context) error {
 		}
 		return err
 	}
+	if apiequality.Semantic.DeepEqual(gotAppRev.Spec, appRev.Spec) {
```

Comment:
please also compare the annotations and labels from metadata

Reply:
ok